app/experimenter/legacy-ui/rapid/components/experiments/ResultsTable.tsx

```diff
@@ -0,0 +1,270 @@
+import React from "react";
+import ReactTooltip from "react-tooltip";
+
+import {
+  SIGNIFICANCE_TIPS,
+  RESULT_COLUMN_TIPS,
+} from "experimenter-rapid/components/experiments/tooltips";
+import { featureOptions } from "experimenter-rapid/components/forms/ExperimentFormOptions";
+import { AnalysisPoint, ExperimentData } from "experimenter-types/experiment";
+
+enum Significance {
+  POSITIVE,
+  NEGATIVE,
+  NEUTRAL,
+}
+
+// A mapping of metric label to relevant statistic. This is
+// used to see which statistic will be used for each metric.
+const resultsMetricsMap = {
+  retained: "binomial",
+  search_count: "mean",
+  identity: "count",
+};
+
+const setPopulationPercentages = (results, totalPopulation) => {
+  Object.keys(results).forEach((branch) => {
+    const populationCount = results[branch]["identity"]["point"];
+    const populationPercent = Math.round(
+      (populationCount / totalPopulation) * 100,
+    );
+    results[branch]["identity"]["percent"] = populationPercent;
+  });
+};
+
+const getFeatureNameMappings = () => {
+  // Create feature-name mappings.
+  const featureNameMappings = {};
+  featureOptions.forEach((featureRow) => {
+    featureNameMappings[featureRow["value"]] = featureRow["name"];
+  });
+  return featureNameMappings;
+};
+
+const getResultMetrics = (featureData) => {
+  // This is used in conjunction with resultsData as an ordered
+  // list of metrics to display in the table from left to right
+  const resultsMetricsList = [
+    {
+      value: "retained",
+      name: "2-Week Browser Retention",
+      tooltip: RESULT_COLUMN_TIPS.RETENTION,
+    },
+    {
+      value: "search_count",
+      name: "Daily Mean Searches Per User",
+      tooltip: RESULT_COLUMN_TIPS.SEARCH,
+    },
+    {
+      value: "identity",
+      name: "Total Users",
+      tooltip: RESULT_COLUMN_TIPS.USER_COUNT,
+    },
+  ];
+
+  const featureNameMappings = getFeatureNameMappings();
+  featureData.forEach((feature) => {
+    const featureMetricID = `${feature}_ever_used`;
+    resultsMetricsMap[featureMetricID] = "binomial";
+    resultsMetricsList.unshift({
+      value: featureMetricID,
+      name: `${featureNameMappings[feature]} Conversion`,
+      tooltip: RESULT_COLUMN_TIPS.CONVERSION,
+    });
+  });
+
+  return { resultsMetricsList, resultsMetricsMap };
+};
+
+const computeSignificance = (lower, upper) => {
+  if (lower < 0 && upper < 0) {
+    return Significance.NEGATIVE;
+  } else if (lower > 0 && upper > 0) {
+    return Significance.POSITIVE;
+  } else {
+    return Significance.NEUTRAL;
+  }
+};
```

Comment:
Oh I just realized, in rapid we don't have ts strict mode enabled but in nimbus we do, maybe it'd be worth while to explicitly type annotate all these functions so they're more easily migratable to nimbus ui?

Reply:
I've updated the signatures for all the functions.